.gitlab-ci-check-shell-format.yml

```diff
@@ -0,0 +1,72 @@
+# .gitlab-ci-check-shell-format.yml
+#
+# This gitlab-ci file runs the shfmt code formatter https://github.com/mvdan/sh
+# on the shell code in the repository, and fails the pipeline if the code is
+# not formatted correctly
+#
+# Add it to the project in hand through Gitlab's include functionality
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-check-shell-format.yml'
+#
+# By default, it will check all files with "*.sh" extension. To change this,
+# override before_script defining your custom SHELL_SCRIPTS variable. For
+# example:
+#
+#  before_script:
+#    - SHELL_SCRIPTS=$(find mender-convert* modules configs scripts
+#            -type f
+#            -not -name "*.md"
+#            -not -path "scripts/linkbot/*"
+#            -not -wholename "scripts/test/*")
+#
+
+stages:
+  - test
+
+test:check-shell-formatting:
+  stage: test
+  needs: []
+  image: alpine:3.13
+  before_script:
+    - SHELL_SCRIPTS=$(find . -type f -name "*.sh")
```

Comment:
I suppose you could also find scripts whose first line is `#!/bin/(ba)?sh`.

Reply:
Good idea!
Let's do both

Reply:
We could, but then we should also accept `#! /usr/bin/env bash` (used in mender-convert) and we would potentially be missing files that are meant to be sourced.

> Let's do both

This is a bad idea from a template point of view, I think. A template should define "an standard" (for example based on file extension, or based on the shebang) but not both. Then a non-standard project can do whatever they want defining this variable.

Reply:
Thanks for your suggestions but I'll leave it as-is. Basically because I already have it working and tested :wrench: